nanoFramework.Device.Bluetooth/BluetoothSignalStrengthFilter.cs

```diff
@@ -16,38 +19,187 @@ public class BluetoothSignalStrengthFilter
+            }
+
+            if (scan != null)
+            {
+                scan.active = true;  // flag entry as active so not purged
+
+                if (!inRange)
+                {
+                    if (args.RawSignalStrengthInDBm < OutOfRangeThresholdInDBm)
+                    {
+                        // Completely out of range, ignore
+                        DeleteScanEntry(args.BluetoothAddress);
+                        return false;
+                    }
+
+                    // In between in and out of range thresholds, check time out of range
+                    if (scan.inRange)
+                    {
+                        // If previously in range and now out
+                        // then set date time for time out of range
+                        AddOrReplaceScanEntry(args.BluetoothAddress, args.RawSignalStrengthInDBm, inRange);
+                    }
+                    else
+                    {
+                        // If previously moved out of range then check timer and still out of range.
+                        if ((scan.outRangeTime + OutOfRangeTimeout) < DateTime.UtcNow)
+                        {
+                            // Moved out of range for time out period
+                            // Remove scan
+                            DeleteScanEntry(args.BluetoothAddress);
+                            return false;
+                        }
+                    }
+                }
+
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Called every 5 minutes to purge scan hash.
+        /// </summary>
+        /// <param name="state"></param>
+        private void scanCheckCallback(object state)
+        {
+            lock (_scanResultsLock)
+            {
+                ArrayList removeList = new ArrayList();
+
+                foreach (DictionaryEntry item in _scanResults)
+                {
+                    if (!((ScanItem)item.Value).active)
+                    {
+                        removeList.Add(item.Key);
+                    }
+                    else
+                    {
+                        ((ScanItem)item.Value).active = false;
+                    }
+                }
+
+                // Now delete expired items
+                foreach (ulong addressKey in removeList)
+                {
+                    DeleteScanEntry(addressKey);
+                }
+            }
+        }
+
+        private ScanItem FindScanEntry(UInt64 address)
+        {
+            if (_scanResults.Contains(address))
+            {
+                return (ScanItem)_scanResults[address];
+            }
+            return null;
+        }
+
+        private ScanItem AddOrReplaceScanEntry(UInt64 address, short Rssi, bool InRange)
+        {
+            ScanItem item = new()
+            {
+                rssi = Rssi,
```

Comment:
And see my previous comment, the class should have the public properties starting with upper case, and the method with lower case, so here, it will be revered at the end `Rssi = rssi;`

Reply:
Done, I noticed when I looked at last change :-)